packages/govuk-frontend/src/govuk/components/character-count/character-count.mjs

```diff
@@ -88,7 +88,11 @@ export class CharacterCount extends GOVUKFrontendComponent {
         $textarea instanceof HTMLInputElement
       )
     ) {
-      return this
+      throw new ElementError($textarea, {
+        componentName: 'Character count',
+        identifier: '.govuk-js-character-count',
```

Comment:
Linking with https://github.com/alphagov/govuk-frontend/pull/4263#pullrequestreview-1644001470

For these "module child" selectors it could be brilliant to show a scope that the browser can inspect

```suggestion
        identifier: '.govuk-js-character-count',
        scope: $module,
```

Reply:
🤔 Yeah, that's probably the component most likely to be on a page multiple times. We recommend to [start with a single question per page](https://design-system.service.gov.uk/patterns/question-pages/#start-by-asking-one-question-per-page) so that limits the risks. It doesn't completely avoids the possibility of multiple of those on the page (especially if we consider the radios/checkboxes potentially revealing fields). That said, there shouldn't be too many of them on a given page that it's tricky for developers to find back the one that's wonky (either by looking carefully at the page or breaking on error in the devtools). I would say it's not something we should not worry too hard about at the start. We can always dig deeper if we receive feedback.

If we have a simple solution, though, not against adding extra information, but it might be tricky to add that consistently. The element with `data-module` does not really have anything identifiable usually. Here, the `$textarea` could provide an `id` (but only if it is on the page so nothing if it doesn't exist).

Reply:
Whoops, there wasn't your second message when I started writing mine, sorry. How would users access the `$module` you propose passing as `$scope` (it's fine if that part still needs looking into)? Do errors log their extra fields? the `cause` maybe, in browsers that support it (that'd be neat)?

Reply:
One to investigate. Would be amazing if you could hover over `scope` and it highlighted the element

Can leave them off for now if we don't need it

Reply:
That'd speed debugging indeed, but let's see if people come back needing more pointers to where things happen as I think that'd need a good bit of investigation 😊 Had a quick glance at `cause`, Firefox logs it (only if the element itself, if an object, you get `Object {...}`, not expandable), not Chrome, nor Safari (at the time of this message). We could always override `toString` to log before the message gets computed for display in the console, but seems quite hacky anyway 😆

Reply:
Sorry, not the clearest blurb. I'll (try to) clarify:

- I'd prefer leaving it off until we have the need for it as it'd require more investigation to make it work
- Got curious and gave a quick check at where Error's `cause` are logged (no luck there) and tried a quick hack by overriding `toString` to log before the error message gets computed (can log the element, but a bit hacky).

Reply:
No don't worry, the confused face was wondering why `cause` wasn't logged

One for another day (and don't want to be Chrome specific) but some nice ideas in:
https://developer.chrome.com/docs/devtools/console/log/